doc/development/06_ca_manager.md

```diff
@@ -0,0 +1,978 @@
+        actor: &Actor
+    ) -> KrillResult<()> { ... }
+
+    /// Updates a parent of a CA, this can be used to update the service uri and/or
+    /// identity certificate for an existing parent.
+    pub async fn ca_parent_update(
+        &self,
+        handle: Handle,
+        parent: ParentHandle,
+        contact: ParentCaContact,
+        actor: &Actor,
+    ) -> KrillResult<()> { ... }
+
+    /// Returns the parent statuses for this CA
+    pub async fn ca_parent_statuses(&self, ca: &Handle) -> KrillResult<ParentStatuses> { ... }
+
+    /// Refresh all CAs:
+    /// - process all CAs in parallel
+    /// - process all parents for CAs in parallel
+    ///    - send pending requests if present, or
+    ///    - ask parent for updates and process if present
+    pub async fn cas_refresh_all(&self, actor: &Actor) { ... }
+
+    /// Synchronizes a CA with one of its parents:
+    ///   - send pending requests if present; otherwise
+    ///   - get and process updated entitlements
+    ///
+    /// Note: if new request events are generated as a result of processing updated entitlements
+    ///       then they will trigger that this synchronization is called again so that the pending
+    ///       requests can be sent.
+    pub async fn ca_sync_parent(&self, handle: &Handle, parent: &ParentHandle, actor: &Actor) -> KrillResult<()> { ... }
+}
+```
+
+
+CA as Parent Related Functions
+------------------------------
+
+The following functions are used to manage children of CAs.
+
+```rust
+/// # CAs as children
+///
+impl CaManager {
+    /// Adds a child under a CA. The 'service_uri' is used here so that
+    /// the appropriate `ParentCaContact` can be returned. If the `AddChildRequest`
+    /// contains resources not held by this CA, then an `Error::CaChildExtraResources`
+    /// is returned.
+    pub async fn ca_add_child(
+        &self,
+        ca: &Handle,
+        req: AddChildRequest,
+        service_uri: &uri::Https,
+        actor: &Actor,
+    ) -> KrillResult<ParentCaContact> { ... }
+
+    /// Show details for a child under the TA.
+    pub async fn ca_show_child(
+        &self,
+        ca: &Handle,
+        child: &ChildHandle
+    ) -> KrillResult<ChildCaInfo> { ... }
+
+    /// Gets an RFC8183 Parent Response for the child.
+    pub async fn ca_parent_response(
+        &self,
+        ca: &Handle,
+        child_handle: ChildHandle,
+        tag: Option<String>,
+        service_uri: &uri::Https,
+    ) -> KrillResult<rfc8183::ParentResponse> { ... }
+
+    /// Update a child under this CA. The submitted `UpdateChildRequest` can contain a
+    /// new `IdCert`, or `ResourceSet`. If both are updated in a single update, then
+    /// an `Error::CaChildUpdateOneThing` is returned. When resource entitlements are updated,
+    /// the existing entitlements are replaced by the new value - i.e. this is not a delta
+    /// and it affects all INR types. Setting resource entitlements beyond the resources
```

Comment:
Novice question: what is INR?

Reply:
Internet Number Resource. I can expand this :D